xls/modules/dbe/lz4_decoder.x

```diff
@@ -0,0 +1,618 @@
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:3, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:3, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:3, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:3, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:3, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:14, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:14, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:15, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:14, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:14, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:15, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:14, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:14, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:9, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:2, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:15, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:14, mark: Mark::NONE},
+    PlainData{is_mark: false, data: uN[4]:8, mark: Mark::NONE},
+    PlainData{is_mark: true, data: uN[4]:0, mark: Mark::END}
+];
+
+#[test_proc]
+proc test_simple {
```

Comment:
Suggest adding a couple of more tests to target edge conditions
1. "overlap match"
2. History buffer wraparouond
3. offset zero

Reply:
Ack.